### ISSUE TITLE
Failed database creation doesn't clean up files.

### ISSUE THREAD

**Issue by MLidstrom** [Starcounter/Starcounter#3268]
If a database creation fails - for example trying to create a database on a port that is already in use - it fails with "ScErrHandlerAlreadyRegistered (SCERR13003): This handler has already been registered. Can't register URI handler "GET /sc/db/{?}/{?}" on port 8080. This handler is already registered. Version: 2.1.1048. Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR13003."

The database wont be listed since it wasn't created, but files exists in the Starcounter directory which means that the second attempt with a new port number fails with "ScErrUnexpectedCommandException (SCERR10016): Creating database myDatabase failed due to an unexpected problem."

**Comment by per-samuelsson** [comment]
@MLidstrom 

Thanks for reporting this. I also noticed this when I tested, so its easy reproducible.

@andwah?

**Comment by miyconst** [comment]
+1

@andwah I think that:

- Administrator should check whether port is available before creating any files.
- Administrator should ask "The directory *myDatabase* is not empty, do you want to remove it?" if by any chance there left some old or corrupted files.

**Comment by per-samuelsson** [comment]
> Administrator should check whether port is available before creating any files.

We have a really old issue somewhere that we want to "pick next available port" when creating databases, but it seems it never got any real love.

**Comment by miyconst** [comment]
Here is the issue: https://github.com/Starcounter/Starcounter/issues/2557

But it is a bit different. It is when we start an app from Visual Studio, there is no database yet, and the `8080` port is occupied by some other program.

**Comment by andwah** [comment]
> Administrator should check whether port is available before creating any files.

Does it mean that we should block so databases can not use the same port number? 

Then we may change so we have a range of portnumbers in the configuration that we can pick from rather just to specify one port (8080) in the configuration.

**Comment by miyconst** [comment]
>Does it mean that we should block so databases can not use the same port number?

No, I didn't mean that. There might be other databases using same port, but not running at the moment.

**Comment by andwah** [comment]
What can be the downside to assure that no databases has the same port number?  

If we make sure that ports are unique then we will not have a problems with ports are occupied when starting databases.
